mountinfo/mounted_linux.go

```diff
@@ -7,16 +7,40 @@ import (
 	"golang.org/x/sys/unix"
 )
 
+// MountedFast is a method of detecting mount points really fast.
+// It works for all kinds of mounts (incl. bind mounts) but requires
+// a recent (v5.6+) Linux kernel. A mount-point check is guaranteed to
+// be a mount-point or not only when sure is true. When sure is false,
+// the caller needs to check for other methods (eg: parse /proc/mounts)
+// to successfully determine if it is a mount-point.
+func MountedFast(path string) (mounted, sure bool, err error) {
+	// root is always mounted
+	if path == string(os.PathSeparator) {
+		return true, true, nil
+	}
+
+	nPath, err := normalizePath(path)
+	if err != nil {
+		return false, false, err
+	}
+
+	return mountedByOpenat2(nPath)
+}
+
 // mountedByOpenat2 is a method of detecting a mount that works for all kinds
 // of mounts (incl. bind mounts), but requires a recent (v5.6+) linux kernel.
-func mountedByOpenat2(path string) (bool, error) {
+// A mount-point check is guaranteed to be a mount-point or not only when 
+// sure is true. When sure is false, the caller needs to check for other '
+// methods (eg: parse /proc/mounts) to successfully determine if it is a
+// mount-point.
+func mountedByOpenat2(path string) (mounted, sure bool, err error) {
```

Comment:
How about returning an exported error instead of the sure flag?

Apparently in all error cases, sure=false. So returning an error wouldn't override the original errors.

Generally speaking Go code checks `err` first and doesn't consume other returning values `if err != nil`. I want to follow the convention here.

Reply:
@kolyshkin Do you have thoughts as you suggested this? I think it boils down to preference.

Reply:
I agree with @kzys here -- no need to change mountedByOpenat2 signature.

Basically, you need to take `mounted` from `mountinfo/mounted_linux.go`, add `sure bool` to it, and remove the fallback to mountinfo parsing. That's all what's needed.